### PR TITLE
added option to cache_unique_id

### LINKED ISSUE #210: Jupyter Notebooks causes unique_id caching
When using Juptyer Notebooks in combination with gokart.build, I found that self.task_unique_id gets cached unless I restart the kernel or drop the task class from memory.

This is an issue because if a task's dependencies' params are updated without clearing memory, the task will not generate a new task_unique_id (hash_id), meaning it won't run with the changed params.

https://github.com/m3dev/gokart/blob/575207f5d348b8c1afbb6d1af3777c67230e15ed/gokart/task.py#L261-L263

I got around this by forcing the hash_id to be generated each time, but I do not believe this to be ideal.
```python
def make_unique_id(self):    
    return self._make_hash_id() 
```

### PR DESCRIPTION
**Pull request by TaylerUva**
added option to cache_unique_id
maintains backwards compatibility
fixes #210

### CODE REVIEW

**Comment by vaaaaanquish** [comment]
@TaylerUva Thx. 🆒 
Can you write unittest?

**Comment by hirosassa** [comment]
Thanks for your contribution!
Your change looks good.
Please fix formatting, and add UT for this change.

**Review by hirosassa — CHANGES_REQUESTED** [review]
Thanks for your rapid response!
I added a comment.

File: .gitignore [hunk 1]
```diff
@@ -112,3 +112,9 @@ examples/resources
 
 # poetry
 dist
+
+# Development
+.vscode
+
+# macOS Files
+**/.DS_Store
```
[the remark this answers is not in the document]
Comment (TaylerUva) [reply .gitignore:120]: I can see your point with not adding .vscode to gitignore. However I believe ignoring operating system files is pretty standard and should be implemented. I would even suggest ignoring standard os files for all operating systems.

[example os gitignore generated from gitignore.io](https://www.toptal.com/developers/gitignore/api/windows,macos,linux)

[the remark this answers is not in the document]
Comment (TaylerUva) [reply .gitignore:120]: Did some research on this topic and I agree now. Thanks for the feedback!

**Comment by hirosassa** [comment]
Looks great! Thank you @TaylerUva !

**Review by hirosassa — APPROVED** [review]

**Comment by vaaaaanquish** [comment]
Thanks. 🆒

**Review by vaaaaanquish — APPROVED** [review]

File: test/test_cache_unique_id.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        self.assertEqual(output1, output2)
+
+    def test_cache_unique_id_false(self):
+        _DummyTaskDep.param = luigi.Parameter(default='original_param')
+
+        output1 = gokart.build(_DummyTask(cache_unique_id=False), reset_register=False)
+
+        _DummyTaskDep.param = luigi.Parameter(default='updated_param')
+        output2 = gokart.build(_DummyTask(cache_unique_id=False), reset_register=False)
+        self.assertNotEqual(output1, output2)
```
Comment (Hi-king) [inline test/test_cache_unique_id.py:48]: Comprehensive tests! 🆒

**Review by Hi-king — APPROVED** [review]

**Comment by Hi-king** [comment]
Merged! Thx for your great contribution @TaylerUva 

And for reviews @vaaaaanquish @hirosassa

**Pull request merged**